pkg/controller/controlplane/valuesprovider_test.go

```diff
@@ -165,6 +200,9 @@ var _ = Describe("Valueprovider Reconcile", func() {
 			Expect(err).NotTo(HaveOccurred())
 
 			cluster := &controller.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: ns.Name,
+				},
```

Comment:
No need to set the name here as the cluster resource is not applied.

Reply:
I am using to get onmetal client in valuesprovider.go

onmetalClient, namespace, err := onmetal.GetOnmetalClientAndNamespaceFromCloudProviderSecret(ctx, vp.Client(), **cluster.ObjectMeta.Name**)